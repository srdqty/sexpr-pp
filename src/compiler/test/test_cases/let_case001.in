((lambda (x y z) (+ x y z)) 1 2 3)
((lambda (x y z) (+ x y z)) 4)
((lambda (x y z) (+ x y z)) 4 5)
((lambda (head . tail) (cons head tail)) 1 2 3 4 5)
