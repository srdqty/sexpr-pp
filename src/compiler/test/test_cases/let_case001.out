(Let 3 ((Constant 1) (Constant 2) (Constant 3)) (App (GlobalRef 0) (ArgCount 3) (Args (ShallowRef 0) (ShallowRef 1) (ShallowRef 2))))
(App (FunFixArity (Arity = 3) (App (GlobalRef 0) (ArgCount 3) (Args (ShallowRef 0) (ShallowRef 1) (ShallowRef 2)))) (ArgCount 1) (Args (Constant 4)))
(App (FunFixArity (Arity = 3) (App (GlobalRef 0) (ArgCount 3) (Args (ShallowRef 0) (ShallowRef 1) (ShallowRef 2)))) (ArgCount 2) (Args (Constant 4) (Constant 5)))
(App (FunVarArity (Arity >= 1) (App (GlobalRef 0) (ArgCount 2) (Args (ShallowRef 0) (ShallowRef 1)))) (ArgCount 5) (Args (Constant 1) (Constant 2) (Constant 3) (Constant 4) (Constant 5)))
